src/pages/filter/ui/panels/function-panel/components/inheritance-mode.tsx

```diff
@@ -1,201 +1,230 @@
-                  variants.map(variant => variant[0]),
-                )
-              }}
-            >
-              {t('general.selectAll')}
-            </p>
-            <p
-              className="text-12 leading-14px text-blue-bright cursor-pointer"
-              onClick={() => setFieldValue('variants', [])}
-            >
-              {t('general.clearAll')}
-            </p>
-          </div>
+    functionPanelStore.fetchStatFunc(
+      FuncStepTypesEnum.InheritanceMode,
+      JSON.stringify({ problem_group: problemGroupValues }),
+    )
+  }
+
+  const handleSelectAllVariant = () => {
+    if (filteredVariants.length === 0) return
+
+    const variantsGroup = filteredVariants.map(variant => variant[0])
+
+    functionPanelStore.setCachedValues<IInheritanceModeCachedValues>(
+      FuncStepTypesEnum.InheritanceMode,
+      {
+        conditions: { problem_group: problemGroupValues },
+        variants: variantsGroup,
+      },
+    )
+  }
+
+  const handleResetAllFieldsLocally = () => {
+    if (problemGroupValues.length === 0) return
+
+    functionPanelStore.clearCachedValues(FuncStepTypesEnum.InheritanceMode)
+  }
+
+  const handleResetVariantsLocally = () => {
+    if (variantsValues.length === 0) return
 
-          {variants.map(variant => {
-            if (variant[1] === 0) {
-              return <Fragment />
-            }
-
-            return (
-              <div key={variant[0]} className="flex items-center mt-4">
-                <Checkbox
-                  checked={variantsValues.includes(variant[0])}
-                  onChange={e => {
-                    const value = e.target.checked
-                      ? [...variantsValues, variant[0]]
-                      : variantsValues.filter(item => item !== variant[0])
-
-                    setFieldValue('variants', value)
-                  }}
-                />
-                <span className="text-14 leading-16px ml-2">{variant[0]}</span>
-                <span className="text-14 leading-16px text-grey-blue ml-1">{`(${variant[1]})`}</span>
-              </div>
-            )
-          })}
-
-          {variants.length === 0 && (
-            <div className="flex justify-center w-full mt-2 text-14 text-grey-blue">
-              Out of choice. Select problem group.
-            </div>
-          )}
-        </Form>
-
-        <PanelButtons
-          selectedFilterName={filterStore.selectedGroupItem.name}
-          selectedFilterGroup={filterStore.selectedGroupItem.vgroup}
-          onSubmit={submitForm}
-          resetForm={resetForm}
-          resetFields={handleResetFields}
-          disabled={variantsValues.length === 0}
-        />
-      </React.Fragment>
+    functionPanelStore.clearCachedValues(
+      FuncStepTypesEnum.InheritanceMode,
+      'variants',
     )
-  },
-)
+  }
+
+  // to avoid displaying this data on the another func attr
+  useEffect(() => {
+    return () => filterStore.resetStatFuncData()
+  }, [])
+
+  return (
```

Comment:
again, too much code in JSX. Try to split it

Reply:
fixed in https://github.com/ForomePlatform/Anfisa-React-Client/pull/444